plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -664,6 +665,82 @@ describe('KubernetesProxy', () => {
     expect(response.status).toEqual(200);
   });
 
+  describe('when server uses TLS', () => {
+    let httpsRequest: jest.SpyInstance;
+    beforeAll(() => {
+      httpsRequest = jest.spyOn(
+        // this is pretty egregious reverse engineering of msw.
+        // If the SetupServerApi constructor was exported, we wouldn't need
+        // to be quite so hacky here
+        (worker as any).interceptor.interceptors[0].modules.get('https'),
+        'request',
+      );
```

Comment:
I know you copy-pasted this (including the apologetic comment) from here:

https://github.com/backstage/backstage/blob/35f05b0f32bfe072a387c9c989a675f8a5021e98/plugins/kubernetes-backend/src/service/KubernetesFetcher.test.ts#L642-L648

If we're going to reuse this, can we extract it to a helper function like `httpsRequestSpy(worker)` and put it in the backend-test-utils package alongside the existing `setupRequestMockHandlers` [here](https://github.com/backstage/backstage/tree/master/packages/backend-test-utils/src/msw)? Then we can encapsulate the ugly hack and hopefully fix it in a single place in the future.

Reply:
Done !